### PR TITLE
Refactor get notifications and search updater

### PR DESCRIPTION
**Pull request by i-like-robots**
Continuing from #540 

- Refactors get notifications module to use fetch and promises
- Refactors notifications handler (nee search updater) to be simpler (courtesy of some ES6 sugar) and provide more consistent logging
- Refactors notifications poll (nee update search) process to be simpler and includes basic spec

Next: re-factor notifications format (and de-dupe, checking splunk we can reingest an item 3 times in a single poll) and then onto the continuous updater.

### CODE REVIEW

**Comment by andygnewman** [comment]
Looks very clear, will let you sort out the linting though.

File: server/processes/search-updater.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
+const REGION = process.env.REGION === 'US' ? 'US' : 'EU';
```
Comment (i-like-robots) [inline server/processes/search-updater.js]: this module should be named `handle-notifications`

File: server/lib/get-notifications.js [hunk 2]
```diff
@@ -1,105 +1,59 @@
...
-		return notifications.links[0].href;
-	} else {
-		return false;
-	}
+	return notifications;
 }
+
+module.exports = function (since, options) {
+	const url = getNotificationsUrl(options);
+	return getNotifications(`${url}?apiKey=${process.env.API_KEY}&feature.blogposts=on&since=${since}`);
```
Comment (i-like-robots) [inline server/lib/get-notifications.js]: assemble the query string in a more elegant way

File: server/poll-notifications.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+const searchUpdater = require('./processes/search-updater');
+const args = require('minimist')(process.argv.slice(2));
+const UPDATE_INTERVAL = args.updateInterval || process.env.UPDATE_INTERVAL || 60 * 1000;
```
Comment (i-like-robots) [inline server/poll-notifications.js]: drop the env var, `parseInt`

File: server/poll-notifications.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+const API_VERSION = parseInt(args.apiVersion) || 1;
```
Comment (i-like-robots) [inline server/poll-notifications.js]: missing radix

**Comment by i-like-robots** [comment]
@andygnewman I've set up some splunk queries to keep an eye this... reckon we're good to 🚢 ?

**Pull request merged**